### PR TITLE
NT: Allow the definition of multiple imagePullSecrets.

### PR DESCRIPTION
**Pull request by jward-bw**
Currently you can only specify a single imagePullSecret in the ContainerImage, for the SolrImage in SolrCloudSpec. This allows a user to define any number of imagePullSecrets, which is required if initContainers or sidecarContainers need different imagePullSecrets to the SolrImage.

### CODE REVIEW

**Comment by jward-bw** [comment]
Sorry for the multiple failed runs, didn't read the docs enough on building the manifests.

**Comment by jward-bw** [comment]
Was I supposed to add `helm/solr-operator/crds/crds.yaml` to my commit? A diff that size smells suspiscious.

File: controllers/util/solr_util.go [hunk 1]
```diff
@@ -520,8 +520,15 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	}
 
 	if solrCloud.Spec.SolrImage.ImagePullSecret != "" {
-		stateful.Spec.Template.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
-			{Name: solrCloud.Spec.SolrImage.ImagePullSecret},
+		if nil != customPodOptions {
+			stateful.Spec.Template.Spec.ImagePullSecrets = append(
+				customPodOptions.ImagePullSecrets,
```
Comment (HoustonPutman) [inline controllers/util/solr_util.go:525]: I think customPodOptions secrets will only be added if the `SolrImage.ImagePullSecret` is provided. You should probably have it added in a separate if statement.

**Review by HoustonPutman — CHANGES_REQUESTED** [review]
Great start! 

Would you mind adding a test for this? And also implement the functionality in the prometheusExporter code as well?

The helm chart change is definitely odd. Are you using the kustomize version specified in `hack/install_dependencies.sh`?

**Comment by jward-bw** [comment]
> Are you using the kustomize version specified in hack/install_dependencies.sh?

Yep that was the issue, I had an older version installed.

> ... implement the functionality in the prometheusExporter code as well?

Done!

> Would you mind adding a test for this?

Sorry this is my first time writing go code, and I'm not familiar with the testing structure. Where should tests for these changes go?

File: controllers/util/solr_util.go (hunk 1 above)
Comment (jward-bw) [reply controllers/util/solr_util.go]: Fixed! Thanks for pointing that out

**Review by HoustonPutman — COMMENTED** [review]
Ok the code looks great!

As for the tests, here is some help:
- [Where we have been specifying customPodOptions test values](https://github.com/apache/lucene-solr-operator/blob/main/controllers/controller_utils_test.go#L535).  
So you should be able to add a list of ImagePullSecrets here to use other places in the tests.
- [An example test that you could add in the imagePullSecrets](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrcloud_controller_test.go#L188)
- [Where you would actually test that they got added to the Pod](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrcloud_controller_test.go#L280)

You can do the same thing for the prometheus exporter tests:
- [An example prom-exp test that you could add in the imagePullSecrets](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrprometheusexporter_controller_test.go#L58)
- [Where you would actually test that they got added to the prom-exp Pod](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrprometheusexporter_controller_test.go#L126)

These tests aren't actually making sure that it runs correctly, it makes sure that the resulting statefulSet object that is created looks as it should. So you just need to make sure that the pod Template in the statefulset has the correct imagePullSecrets section.

**Comment by jward-bw** [comment]
> As for the tests, here is some help:


Thanks so much for the pointers Houston, let me know if I've missed anything.

**Review by HoustonPutman — APPROVED** [review]
Don't think you've missed anything! This looks great, feature & tests. 👌 

Thanks for the contribution! Hopefully the first of many 😉

**Pull request merged**